### DISCUSSION TITLE
Typescript error on shadcn components

### DISCUSSION THREAD

**Discussion by chrisarrow90** [shadcn-ui/ui#3944] [Q&A]
I am trying to use a button as so
`<Button variant="outline" size="icon">`
however I am getting a typescript error on the variant attribute with the following message:

"Type '{ children: Element; variant: string; size: string; }' is not assignable to type 'IntrinsicAttributes & ButtonProps & RefAttributes<HTMLButtonElement>'.
  Property 'variant' does not exist on type 'IntrinsicAttributes & ButtonProps & RefAttributes<HTMLButtonElement>'.ts(2322)"
  
what is the solution to this? Thanks

**Comment by alvinsjoy** [comment]
You might not be importing the button correctly. Check if the import paths are correct.

**Comment by lorynmason** [comment]
I had this same error, not sure if this is the best way of fixing this but in my tsconfig.json I changed  "moduleResolution": "bundler",
 to "moduleResolution": "Node" and now it works

**Comment by wonderkid-web** [comment]
I have same problem with this, and im pretty sure because @types/react v.19 have a bug of this..

so i try to downgrade my @types/react to v.18.3.12 and my error was gone..

soon this bug will be fix,

**Reply by SonuKumar145** [reply]
This issue still persists. But downgrade the @types/react to v.18.3.12 worked.

**Reply by iatomic1** [reply]
> This issue still persists. But downgrade the @types/react to v.18.3.12 worked.

have you found a fix? this doesn't work for me
